pkg/segment/writer/segwriter.go

```diff
@@ -139,6 +139,62 @@ type WipBlock struct {
 	bb                 *bbp.ByteBuffer        // byte buffer pool for HLL byte inserts
 }
 
+type ParsedLogEvent struct {
+	allCnames       []string  // array of all cnames
+	allCvals        [][]byte  // array of all column values byte slices
+	allCvalsTypeLen [][9]byte // array of all column values type and len (3 bytes for strings; 9 for numbers)
+	numCols         uint16    // number of columns in this log record
+	indexName       string
+	rawJson         []byte
+	timestampMillis uint64
+}
+
+func NewPLE() *ParsedLogEvent {
+	return &ParsedLogEvent{
+		allCnames:       make([]string, 0),
+		allCvals:        make([][]byte, 0),
+		allCvalsTypeLen: make([][9]byte, 0),
+		numCols:         0,
+	}
+}
+
+func (ple *ParsedLogEvent) Reset() {
+	ple.allCnames = ple.allCnames[:0]
+	ple.allCvals = ple.allCvals[:0]
+	ple.allCvalsTypeLen = ple.allCvalsTypeLen[:0]
+	ple.numCols = 0
+}
+
+func (ple *ParsedLogEvent) MakeSpaceForNewColumn() {
+	ple.allCnames = append(ple.allCnames, "")
+	ple.allCvals = append(ple.allCvals, nil)
+	ple.allCvalsTypeLen = append(ple.allCvalsTypeLen, [9]byte{})
```

Comment:
nit: lets add a comment here why the 9 bytes

Reply:
Yeah I updated your comment in the type definition:
```
type ParsedLogEvent struct {
	...
	allCvalsTypeLen [][9]byte // array of all column values type and len (3 bytes for strings; 9 for numbers)
	...
}
```

Do you want me to be more clear about why numbers need 9 bytes?